src/components/base/InternalStyledSvg/InternalStyledSvg.tsx

```diff
@@ -0,0 +1,36 @@
+import styled from "styled-components";
+
+import { OakBorderWidthToken, OakCombinedColorToken } from "@/styles";
+import {
+  ResponsiveValues,
+  responsiveStyle,
+} from "@/styles/utils/responsiveStyle";
+import { parseColor } from "@/styles/helpers/parseColor";
+import { parseBorderWidth } from "@/styles/helpers/parseBorderWidth";
+
+export type InternalStyledSvgProps = {
+  $fill?: ResponsiveValues<OakCombinedColorToken>;
+  $stroke?: ResponsiveValues<OakCombinedColorToken>;
+  $strokeWidth?: ResponsiveValues<OakBorderWidthToken>;
+};
+
+/**
+ * A styled SVG element with props to apply design tokens to the fill and stroke.
+ */
+export const InternalStyledSvg = styled.svg<InternalStyledSvgProps>`
+  ${responsiveStyle<InternalStyledSvgProps, OakCombinedColorToken>(
+    "fill",
+    (props) => props.$fill,
+    parseColor,
+  )}
```

Comment:
i feel liket his slightly breaks our convention that "styles" live in the style/utils folder and then they are imported into the base components; rather then putting the style logic inside the base components file